artifactory/commands/helm/flags.go

```diff
@@ -0,0 +1,251 @@
+		arg := args[i]
+		if handleFlagWithValue(flags, &i, arg, args) || handleFlagWithEquals(flags, arg) {
+			continue
+		}
+		collectHelmArg(flags, i, arg, args)
+	}
+}
+
+// handleFlagWithValue handles flags with separate values (--flag value)
+func handleFlagWithValue(flags *helmFlags, i *int, arg string, args []string) bool {
+	if *i+1 >= len(args) {
+		return false
+	}
+	nextArg := args[*i+1]
+	switch arg {
+	case "--server-id":
+		if flags.serverId == "" {
+			flags.serverId = nextArg
+		}
+		*i++
+		return true
+	case "--username", "--user":
+		if flags.username == "" {
+			flags.username = nextArg
+		}
+		*i++
+		return true
+	case "--password":
+		if flags.password == "" {
+			flags.password = nextArg
+		}
+		*i++
+		return true
+	case "--url":
+		if flags.url == "" {
+			flags.url = nextArg
+		}
+		*i++
+		return true
+	case "--build-name", "--build-number", "--project", "--module":
+		*i++
+		return true
+	case "--skip-login":
+		return true
+	}
+	return false
+}
+
+// handleFlagWithEquals handles flags with equals format (--flag=value)
+func handleFlagWithEquals(flags *helmFlags, arg string) bool {
+	if !strings.Contains(arg, "=") {
+		return false
+	}
+
+	parts := strings.SplitN(arg, "=", 2)
+	if len(parts) != 2 {
+		return false
+	}
+
+	flagName := parts[0]
+	flagValue := parts[1]
+
+	switch flagName {
+	case "--server-id":
+		if flags.serverId == "" {
+			flags.serverId = flagValue
+		}
+		return true
+	case "--username", "--user":
+		if flags.username == "" {
+			flags.username = flagValue
+		}
+		return true
+	case "--password":
+		if flags.password == "" {
+			flags.password = flagValue
+		}
+		return true
+	case "--url":
+		if flags.url == "" {
+			flags.url = flagValue
+		}
+		return true
+	case "--build-name", "--build-number", "--project", "--module":
+		return true
+	}
+
+	return false
+}
+
+// collectHelmArg collects helm arguments, skipping JFrog-specific flags
+func collectHelmArg(flags *helmFlags, i int, arg string, args []string) {
+	if i < 3 || args[1] != "helm" {
+		return
+	}
+	jfrogFlags := []string{
+		"--build-",
+		"--project=",
+		"--module=",
+		"--server-id",
+		"--url",
+		"--skip-login",
+	}
+	for _, flag := range jfrogFlags {
+		if strings.HasPrefix(arg, flag) {
+			return
+		}
+	}
+	flags.helmArgs = append(flags.helmArgs, arg)
+}
+
+// getHelmCommandName returns the helm command name from os.Args
+func getHelmCommandName() string {
+	return parseHelmFlags().commandName
+}
+
+// getHelmServerId returns the server-id flag value from os.Args
+func getHelmServerId() string {
+	return parseHelmFlags().serverId
+}
+
+// getHelmUsername returns the username flag value from os.Args
+func getHelmUsername() string {
+	return parseHelmFlags().username
+}
+
+// getHelmPassword returns the password flag value from os.Args
+func getHelmPassword() string {
+	return parseHelmFlags().password
+}
+
+// getHelmUrl returns the url flag value from os.Args
+func getHelmUrl() string {
+	return parseHelmFlags().url
+}
```

Comment:
why are these getters required? since the getter itself is not public getters looks redundant and can be removed.